src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -798,6 +799,11 @@ public Change(ChangeType type, Address address, Account? account)
             public Account? Account { get; }
         }
 
+        public AddressAsKey[]? ChangedAddresses()
+        {
+            return _blockCache.Count == 0 ? Array.Empty<AddressAsKey>() : _blockCache.Keys.ToArray();
```

Comment:
Does it makes sense to allocate it as array?

Reply:
What would you use?

Detaches it from block processing; isn't a concurrent Dictionary so can't be accessed safely and the processing it just about to clear it for the next block.

Could pass along the full dictionary and allocate a new one; but using the updated data caused more cache misses. Would probably need to add `UpdateValueButNotRecency` method for the Lru; as if an account was used in last block, most of them aren't used in the very next block; so want older address that weren't included and still in the txpool to survive.

Reply:
If there is a good time to release it then it can be ArrayPoolList

Reply:
Ohhh, good point